client/tests/integration/restart_peer.rs

```diff
@@ -57,7 +57,9 @@ fn restarted_peer_should_have_the_same_asset_amount() -> Result<()> {
             .expect("Asset not found");
         assert_eq!(AssetValue::Quantity(quantity), *asset.value());
     }
-
+    peer.iroha.take();
+    // Make sure previous peer has time to shutdown.
+    std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
This is fragile. 

We need some form of message passing so that this peer can wait for the other one to shutdown. 

Consider using an `async` function which blocks until the other peer has shutdown.